src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtOpeningTimeService.java

```diff
@@ -0,0 +1,75 @@
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.dts.fact.entity.Court;
+import uk.gov.hmcts.dts.fact.entity.CourtOpeningTime;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.OpeningTime;
+import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class AdminCourtOpeningTimeService {
+    private final CourtRepository courtRepository;
+
+    @Autowired
+    public AdminCourtOpeningTimeService(final CourtRepository courtRepository) {
+        this.courtRepository = courtRepository;
+    }
+
+    public List<OpeningTime> getCourtOpeningTimesBySlug(final String slug) {
+        return courtRepository.findBySlug(slug)
+            .map(c -> c.getCourtOpeningTimes()
+                .stream()
+                .map(CourtOpeningTime::getOpeningTime)
+                .map(OpeningTime::new)
+                .collect(toList()))
+            .orElseThrow(() -> new NotFoundException(slug));
+    }
+
+    public List<OpeningTime> updateCourtOpeningTimes(final String slug, final List<OpeningTime> openingTimes) {
+        final Court courtEntity = courtRepository.findBySlug(slug)
+            .orElseThrow(() -> new NotFoundException(slug));
+        return saveNewOpeningTimes(courtEntity, openingTimes);
+    }
+
+    private List<OpeningTime> saveNewOpeningTimes(final Court courtEntity, final List<OpeningTime> openingTimes) {
+        List<uk.gov.hmcts.dts.fact.entity.OpeningTime> openingTimesEntity = getNewOpeningTimesEntity(openingTimes);
+        List<CourtOpeningTime> courtOpeningTimesEntity = getNewCourtOpeningTimesEntity(courtEntity, openingTimesEntity);
+
+        if (courtEntity.getCourtOpeningTimes() == null) {
+            courtEntity.setCourtOpeningTimes(courtOpeningTimesEntity);
+        } else {
+            courtEntity.getCourtOpeningTimes().clear();
+            courtEntity.getCourtOpeningTimes().addAll(courtOpeningTimesEntity);
+        }
+        final Court courtWithUpdatedOpeningTime = courtRepository.save(courtEntity);
+
+        return courtWithUpdatedOpeningTime.getCourtOpeningTimes()
+            .stream()
+            .map(CourtOpeningTime::getOpeningTime)
+            .map(OpeningTime::new)
+            .collect(toList());
+    }
+
+    private List<uk.gov.hmcts.dts.fact.entity.OpeningTime> getNewOpeningTimesEntity(final List<OpeningTime> openingTimes) {
+        return openingTimes.stream()
+            .map(o -> new uk.gov.hmcts.dts.fact.entity.OpeningTime(o.getType(), o.getTypeCy(), o.getHours()))
+            .collect(toList());
+    }
+
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    private List<CourtOpeningTime> getNewCourtOpeningTimesEntity(final Court court, final List<uk.gov.hmcts.dts.fact.entity.OpeningTime> openingTimes) {
+        final List<CourtOpeningTime> courtOpeningTimes = new ArrayList<>();
```

Comment:
Did you say on the other PR you preferred to have this as `emptyList`? Or would it not equate to the same thing? 

`final List<CourtOpeningTime> courtOpeningTimes = emptyList()`

Reply:
I think for declaration, new ArrayList() is clearer. This one is used initialise a list before adding anything it. I know they are the same but I prefer calling the new operator explicitly in this case :)